modules/core/src-3/NewtypeValidated.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ *  Copyright 2021-2024 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s
+
+abstract class NewtypeValidated[A] extends HasId { self =>
+  opaque type Type = A
+
+  def apply(a: A): Either[String, Type]
+
+  def unsafeApply(a: A): Type = a
```

Comment:
**question:** (for discussion): should we make this throw in case of invalid values? We could also have both variants available, although the "just wrap, not questions asked" one should be so obscure that nobody will accidentally use it.

Reply:
Changed this to throw an exception. Decided against having an "obscure, unsafe-not-even-throwing" method. If anyone needs that behavior, they could just catch the exception and ignore it?

Reply:
yeah, I think that's fair